lib/pbench/agent/tool_meister.py

```diff
@@ -1518,6 +1533,39 @@ def end_tools(self, data):
                     name,
                 )
                 failures += 1
+
+        # Remove persistent tool temporary working directory on remote host
+        directory = data["directory"]
+        tool_dir = self.directories[directory]
+
+        if self._hostname == self._controller:
+            directory_to_delete = tool_dir
+        else:
+            directory_to_delete = tool_dir.parent
+
+        self.logger.debug(
+            "%s: deleting persistent tool tmp directory %s %s",
+            self._hostname,
+            self._group,
+            directory_to_delete,
+        )
+        unexpected_files = [x for x in directory_to_delete.iterdir() if x.is_file()]
+        if not unexpected_files:
+            try:
+                shutil.rmtree(directory_to_delete)
+            except Exception:
+                self.logger.exception(
+                    f"Failed to remove tmp persistent tool data hierarchy, "
+                    f"{directory_to_delete}"
+                )
+                failures += 1
+        else:
+            self.logger.exception(
+                f"Unexpected files found at tmp persistent tool directory "
+                f"hierarchy, {directory_to_delete}. Files: {unexpected_files}"
+            )
+            failures += 1
+        del self.directories[directory]
         if failures > 0:
             msg = f"{failures} of {tool_cnt} failed stopping persistent tools"
```

Comment:
Are you sure that we want to count the presence of unexpected files or the inability to delete a temporary directory as a "failure"?  They don't result in us failing to obtain the data, so they seem like things that we should just warn the user about.